### PR TITLE
Refactor swap router

### PR DESCRIPTION
**Pull request by defcon022**
## Description

Resolves 
- VEN-1320: Make SwapRouter not upgradable.
- VEN-1321: Add missing swap functions related to BNB.

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: contracts/Swap/RouterHelper.sol [hunk 1]
```diff
@@ -143,26 +155,41 @@ abstract contract RouterHelper is IRouterHelper {
...
-        if (amounts[amounts.length - 1] < amountOutMin) {
-            revert OutputAmountBelowMinimum(amounts[amounts.length - 1], amountOutMin);
-        }
-        TransferHelper.safeTransferFrom(
-            path[0],
-            msg.sender,
-            PancakeLibrary.pairFor(factory, path[0], path[1]),
-            amounts[0]
-        );
+        uint256 WBNBWAmount;
```
Comment (kkirka) [inline contracts/Swap/RouterHelper.sol:158]:
```suggestion
        uint256 WBNBAmount;
```

File: contracts/Swap/SwapRouter.sol [hunk 2]
```diff
@@ -1,6 +1,6 @@
 pragma solidity 0.8.13;
 
-import "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
```
Comment (kkirka) [inline contracts/Swap/SwapRouter.sol:3]: Ownable2Step maybe?

File: contracts/Swap/SwapRouter.sol [hunk 3]
```diff
@@ -16,7 +16,7 @@ import "./interfaces/InterfaceComptroller.sol";
  * @author 0xlucian
  */
 
-contract SwapRouter is Ownable2StepUpgradeable, RouterHelper, IPancakeSwapV2Router {
+contract SwapRouter is Ownable, RouterHelper, IPancakeSwapV2Router {
     using SafeERC20Upgradeable for IERC20Upgradeable;
 
     address public comptrollerAddress;
```
Comment (kkirka) [inline contracts/Swap/SwapRouter.sol:22]: `comptrollerAddress` can't be modified, maybe we could make it immutable for gas savings?

File: contracts/Swap/interfaces/CustomErrors.sol [hunk 4]
```diff
@@ -42,5 +42,8 @@ error IdenticalAddresses();
 ///@notice Error thrown when the trade path[] parameter consists of only 1 token (i.e. path.length<2)
 error InvalidPath();
 
-//@notice Error thrown when invalid vTOken address is passed to swap router.
+///@notice Error thrown when invalid vTOken address is passed to swap router.
 error VTokenNotListed(address vToken);
+
+///@notice Error thrown when addresses path length is less than 2.
+error SwapInvalidAddressesPath(address[] path);
```
Comment (kkirka) [inline contracts/Swap/interfaces/CustomErrors.sol:49]: Is it different from `InvalidPath` defined above? Maybe I miss something but the notices indicate these should be equivalent

File: contracts/Swap/RouterHelper.sol (hunk 1 above)
Comment (defcon022) [reply contracts/Swap/RouterHelper.sol]: Fixed, d066bf93502d88423e3aafe3106bda584cef3103

File: contracts/Swap/SwapRouter.sol (hunk 2 above)
Comment (defcon022) [reply contracts/Swap/SwapRouter.sol]: Fixed, d066bf93502d88423e3aafe3106bda584cef3103

File: contracts/Swap/SwapRouter.sol (hunk 3 above)
Comment (defcon022) [reply contracts/Swap/SwapRouter.sol]: Fixed, d066bf93502d88423e3aafe3106bda584cef3103

File: contracts/Swap/interfaces/CustomErrors.sol (hunk 4 above)
Comment (defcon022) [reply contracts/Swap/interfaces/CustomErrors.sol]: Fixed, d066bf93502d88423e3aafe3106bda584cef3103

**Pull request merged**